From 02f45e043073ec3d3ab0419a20d3b79f8934cc2f Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@st.com>
Date: Mon, 11 Mar 2019 10:31:37 +0100
Subject: [PATCH 189/529] ASoC: wm8994: add LDO cycle-time protection

The minimum cycle time of wm8994 audio codec
LDOx regulator is specified at 36ms.
Add a 40ms delay on regulator enable at resume
to match this requirement.

Change-Id: I15803071a0e641e284d15a16a6056ceeeafd1254
Signed-off-by: Olivier Moysan <olivier.moysan@st.com>
Reviewed-on: https://gerrit.st.com/127625
Reviewed-by: CITOOLS <smet-aci-reviews@lists.codex.cro.st.com>
Reviewed-by: CIBUILD <smet-aci-builds@lists.codex.cro.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 drivers/mfd/wm8994-core.c | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/drivers/mfd/wm8994-core.c
+++ b/drivers/mfd/wm8994-core.c
@@ -185,6 +185,12 @@ static int wm8994_resume(struct device *
 	if (!wm8994->suspended)
 		return 0;
 
+	/*
+	 * LDO1/2 minimum cycle time is 36ms according to codec specification
+	 * Wait before enabling regulator to make sure we fit this requirement
+	 */
+	msleep(40);
+
 	ret = regulator_bulk_enable(wm8994->num_supplies,
 				    wm8994->supplies);
 	if (ret != 0) {
